### PR TITLE
chore: bump version of go and fix lint errors

### PR DESCRIPTION
**Pull request by rkpattnaik780**
PR to update version of `go` and `golangci-lint`.

### Verification Steps

1. Checkout the PR.
2. Update go version in local to 1.18.x
3. Update golangci-lint version in local to 1.48.0
4. Run `make lint`, `make binary`.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation change
- [X] Chore - update dependencies

### CODE REVIEW

File: pkg/cmd/kafka/create/create.go [hunk 1]
```diff
@@ -262,9 +262,13 @@ func runCreate(opts *options) error {
 				return err
 			}
 
-			userQuotaJSON, _ := json.MarshalIndent(userQuota, "", "  ")
-			f.Logger.Debug("Selected Quota object:")
-			f.Logger.Debug(string(userQuotaJSON))
+			userQuotaJSON, lintErr := json.MarshalIndent(userQuota, "", "  ")
+			if lintErr != nil {
```
Comment (wtrocki) [inline pkg/cmd/kafka/create/create.go:266]: lintErr sounds strange. Maybe marshalErr

**Review by wtrocki — APPROVED** [review]

**Comment by jackdelahunt** [comment]
Running go 1.19 and golangci-lint 1.48.0 and got these errors

```
pkg/core/config/file.go:6:2: SA1019: "io/ioutil" has been deprecated since Go 1.16: As of Go 1.16, the same functionality is now provided by package io or package os, and those implementations should be preferred in new code. See the specific function documentation for details. (staticcheck)
	"io/ioutil"
	^
pkg/core/servicecontext/file.go:6:2: SA1019: "io/ioutil" has been deprecated since Go 1.16: As of Go 1.16, the same functionality is now provided by package io or package os, and those implementations should be preferred in new code. See the specific function documentation for details. (staticcheck)
	"io/ioutil"
	^
internal/telemetry/user.go:4:2: SA1019: "io/ioutil" has been deprecated since Go 1.16: As of Go 1.16, the same functionality is now provided by package io or package os, and those implementations should be preferred in new code. See the specific function documentation for details. (staticcheck)
	"io/ioutil"
```

**Comment by rkpattnaik780** [comment]
> Running go 1.19 and golangci-lint 1.48.0 and got these errors
Hi @jackdelahunt . I am upgrading version to 1.18. Have removed the `io/ioutil` package now.

**Comment by jackdelahunt** [comment]
No errors when running `make lint` (some warnings but they are fine I think). 

LGTM :+1:

**Comment by wtrocki** [comment]
Yes. We should add https://github.com/moovweb/gvm as recomendation for users compiling code.

**Pull request merged**